src/util/openfaas/index.js

```diff
@@ -0,0 +1,149 @@
+const NodeCache = require("node-cache");
+const {
+  getFunction,
+  deleteFunction,
+  deployFunction,
+  invokeFunction
+} = require("./faasApi");
+const logger = require("../../logger");
+const { RetryRequestError } = require("../utils");
+
+const FAAS_BASE_IMG =
+  process.env.FAAS_BASE_IMG || "rudderlabs/openfaas-flask:main";
+const FAAS_MAX_PODS_IN_TEXT = process.env.FAAS_MAX_PODS_IN_TEXT || "100";
+const CONFIG_BACKEND_URL =
+  process.env.CONFIG_BACKEND_URL || "https://api.rudderlabs.com";
+
+const functionListCache = new NodeCache();
+const FUNC_LIST_KEY = "fn-list";
+
+const delayInMs = async (ms = 2000) =>
+  new Promise(resolve => setTimeout(resolve, ms));
+
+const callWithRetry = async (fn, count = 0, ...args) => {
```

Comment:
Maybe we can use this library https://www.npmjs.com/package/axios-retry

Reply:
rename to eventually ..
add todo

Reply:
added
`// This api call is only used to check if function is spinned eventually`
 `// TODO: call health endpoint instead of get function to get correct status`